packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1111,22 +1110,36 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return { snapshot, state };
     }
 
+    private replayPendingStates() {
+        // We need to be able to send ops to replay states
+        if (!this.canSendOps()) { return; }
+
+        // Save the old state, reset to false, disable event emit
```

Comment:
Some comments on the reasons for all this state management would be great.
Also should we use helper classes to make sure right stuff happens if exception is thrown from here?

Reply:
added.